crypto/multi_signature.go

```diff
@@ -0,0 +1,89 @@
+package crypto
+
+import (
+	"reflect"
+
+	"github.com/relab/hotstuff"
+	"golang.org/x/exp/slices"
```

Comment:
This can be replaced with just `"slices"` I think. Or it may need the dependency update to get go1.22, but not totally sure. At least if we merge #118 it should work without this `x/exp/slices` package.